src/App.tsx

```diff
@@ -1,61 +1,129 @@
+import React, { useState } from 'react';
+
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo, User } from './types';
+
+const getUserById = (userId: number): User | null =>
+  usersFromServer.find(person => person.id === userId) || null;
+
+const todos: Todo[] = todosFromServer.map(todo => {
+  const user = getUserById(todo.userId);
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+  return { ...todo, user };
+});
+
+const getNewTodoId = (items: Todo[]) => {
+  const maxId = Math.max(...items.map(item => item.id));
+
+  return maxId + 1;
+};
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasSelectedError, setHasSelectedError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleSelectedUserChange = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setSelectedUser(+event.target.value);
+    setHasSelectedError(false);
+  };
+
+  const onAdd = (newTodo: Todo) => {
+    const currentTodo = {
+      ...newTodo,
+      id: getNewTodoId(visibleTodos),
+    };
+
+    setVisibleTodos(currentTodos => [...currentTodos, currentTodo]);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setSelectedUser(0);
+    setHasSelectedError(false);
+    setHasTitleError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasSelectedError(!selectedUser);
+    setHasTitleError(!title);
+
+    if (!selectedUser || !title) {
+      return;
+    }
+
+    onAdd({
+      id: 0,
```

Comment:
In the `onAdd` function, the new todo is initially given an ID of 0. This could potentially conflict with existing todos if not handled properly. Ensure that the `getNewTodoId` function is always called to generate a unique ID for new todos.